### PR TITLE
Docs: Add files in publishing folder to plugin-tools

### LINKED ISSUE #358: Update sign a plugin documentation
Following up on: https://github.com/grafana/plugin-tools/issues/323
We have already updated https://grafana.com/docs/grafana/latest/developers/plugins/publish-a-plugin/sign-a-plugin/#sign-a-public-plugin so we need to also update the current state. 

We should also update this page https://grafana.com/developers/plugin-tools/publish-a-plugin/sign-a-plugin

### PR DESCRIPTION
**Pull request by josmperez**
fixes #358 

Add files in publishing folder of grafana/developers to the plugin-tools repo. 

Note:
- 'distributing-a-plugin.md' is deleted owing to its duplicating content in 'publishing-a-plugin.md'. Please verify that nothing important will be lost.

### CODE REVIEW

File: docusaurus/docs/publish-a-plugin/publishing-and-signing-criteria.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
+id: publishing-and-signing-criteria
+title: Publishing and signing criteria
+sidebar_position: 2
```
Comment (sympatheticmoose) [inline docusaurus/docs/publish-a-plugin/publishing-and-signing-criteria.md:4]:
```suggestion
sidebar_position: 1
```

We've removed the current first item

File: docusaurus/docs/publish-a-plugin/package-a-plugin.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+---
+id: package-a-plugin
+title: Package a plugin
+sidebar_position: 3
```
Comment (sympatheticmoose) [inline docusaurus/docs/publish-a-plugin/package-a-plugin.md:4]:
```suggestion
sidebar_position: 2
```

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md [hunk 3]
```diff
@@ -1,74 +1,88 @@
 ---
 id: sign-a-plugin
 title: Sign a plugin
-sidebar_position: 2
+sidebar_position: 4
```
Comment (sympatheticmoose) [inline docusaurus/docs/publish-a-plugin/sign-a-plugin.md:4]:
```suggestion
sidebar_position: 3
```

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
+---
+id: publish-a-plugin
+title: Publish a plugin
+sidebar_position: 3
```
Comment (sympatheticmoose) [inline docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:4]:
```suggestion
sidebar_position: 4
```

Comment (sympatheticmoose) [reply docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:4]: publish should come after sign imo

File: docusaurus/docs/publish-a-plugin/distribute-a-plugin.md [hunk 5]
```diff
@@ -1,27 +0,0 @@
...
-id: distribute-a-plugin
-title: Distribute a plugin
-sidebar_position: 1
----
-
-To distribute a Grafana plugin either to the community or privately, the plugin must be signed so the Grafana application can verify its authenticity. You can do this with the `@grafana/sign-plugin` package.
-
-:::info
-
-It's not necessary to sign a plugin during development. The [Docker development environment](../get-started/set-up-development-environment.mdx) that is scaffolded with `@grafana/create-plugin` will load the plugin without a signature.
```
Comment (sympatheticmoose) [inline docusaurus/docs/publish-a-plugin/distribute-a-plugin.md:11]: We've lost this information. We should add to the dev environment doc and maybe sign a plugin. WDYT @josmperez ?

File: docusaurus/docs/publish-a-plugin/distribute-a-plugin.md [hunk 6]
```diff
@@ -1,27 +0,0 @@
...
-
-:::
-
-## Initial steps
-
-Before signing a plugin, read the Grafana [plugin publishing and signing criteria](https://grafana.com/docs/grafana/latest/developers/plugins/publishing-and-signing-criteria/) documentation carefully.
-
-`@grafana/create-plugin` adds the necessary commands and github workflows for signing and distributing a plugin via the Grafana plugins catalog.
```
Comment (sympatheticmoose) [inline docusaurus/docs/publish-a-plugin/distribute-a-plugin.md:19]: We've lost this but it was generally a bit vague. A follow up would be to better link the release workflow documentation and sign/publish documentation

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+### Follow our guidelines
+
+Get familiar with our plugin [publishing and signing criteria](./publishing-and-signing-criteria.md)
+
+### Do this for best results
+
+To speed up the time it takes to review your plugin:
+
+- Check that your plugin is ready for review using the [plugin validator](https://github.com/grafana/plugin-validator).
+- Read our [6 tips for improving your Grafana plugin before you publish](https://grafana.com/blog/2021/01/21/6-tips-for-improving-your-grafana-plugin-before-you-publish/).
```
Comment (sympatheticmoose) [inline docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:38]:
```suggestion
```

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+- Refer to [plugin-examples](https://github.com/grafana/grafana-plugin-examples) to review best practices for building your plugin.
+
+## Publish your plugin
```
Comment (sympatheticmoose) [inline docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:41]: Follow up - publish/update differ by so little and take up a lot of screen estate. We should either tab or combine

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md [hunk 9]
```diff
@@ -1,74 +1,88 @@
...
+| **Plugin Level** | **Paid Subscription Required?**                 | **Description**                                                                                                                                                                                                                                                                                                                  |
+| ---------------- | ----------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Private          | No;<br></br>Free of charge                           | <p>You can create and sign a Private plugin for any technology at no charge.</p><p>Private plugins are intended for use on your own installation of Grafana. They may not be distributed to the Grafana community, and they are not published in the Grafana plugin catalog.</p>                                                 |
```
Comment (sympatheticmoose) [inline docusaurus/docs/publish-a-plugin/sign-a-plugin.md:79]: Please update this table with the exact copy from https://grafana.com/legal/plugins/ we should ensure they're identical prior to publishing.

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md [hunk 10]
```diff
@@ -1,74 +1,88 @@
...
+  - signing plugin
 ---
 
-<head>
-  <meta name="robots" content="noindex" />
-</head>
+# Sign a plugin
 
-Grafana requires all plugins to be signed so that we can verify their authenticity with signature verification.
+Grafana requires all plugins to be signed so that we can verify their authenticity with [signature verification](https://grafana.com/docs/grafana/latest/administration/plugin-management#plugin-signatures).
```
Comment (sympatheticmoose) [inline docusaurus/docs/publish-a-plugin/sign-a-plugin.md:16]: Can be a follow up. But theres unnecessary duplication in this and the next sentence

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md [hunk 11]
```diff
@@ -1,74 +1,88 @@
...
 Before you can sign your plugin, you need to decide whether you want to sign it as a _public_ or a _private_ plugin.
 
-To make your plugin publicly available outside of your organization, sign your plugin under a _community_ or _commercial_ signature level. Public plugins are available from the Grafana plugin catalog and can be installed by anyone.
+To make your plugin publicly available outside of your organization, sign your plugin under a _community_ or _commercial_ [signature level](#plugin-signature-levels). Public plugins are available from the [Grafana plugin catalog](https://grafana.com/plugins) and can be installed by anyone.
 
-If you intend to only use the plugin within your organization, sign it under a _private_ signature level.
+If you intend to only use the plugin within your organization, sign it under a _private_ [signature level](#plugin-signature-levels).
 
-## Generate an API key
+## Generate a token
```
Comment (sympatheticmoose) [inline docusaurus/docs/publish-a-plugin/sign-a-plugin.md:26]: follow up - maybe we should look at the exact terminology used here, as you need to generate a cloud access policy and use a token for that policy.

Comment (sympatheticmoose) [reply docusaurus/docs/publish-a-plugin/sign-a-plugin.md:26]:
```suggestion
## Generate an Access Policy token
```

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md [hunk 12]
```diff
@@ -1,74 +1,88 @@
...
+| Community        | No;<br></br>Free of charge                           | <p>You can create, sign, and distribute plugins at no charge, provided that all dependent technologies are open source and not for profit.</p><p>Community plugins are published in the official Grafana catalog, and are available to the entire Grafana community.</p>                                                         |
+| Commercial       | Yes;<br></br>Commercial plugin subscription required | <p>You can create, sign, and distribute plugins with dependent technologies that are closed source or commercially backed. To do so, enter into a Commercial plugin subscription with Grafana Labs.</p><p>Commercial plugins are published on the Grafana plugin catalog, and are available to the entire Grafana community.</p> |
 
-For instructions on how to sign a plugin under the Community and Commercial signature level, refer to Sign a public plugin.
+For instructions on how to sign a plugin under the Community and Commercial signature level, refer to [Sign a public plugin](#sign-a-public-plugin).
```
Comment (sympatheticmoose) [inline docusaurus/docs/publish-a-plugin/sign-a-plugin.md:83]: follow up, these links bring you back up to the previous section (will make a broader comment on this)

**Review by sympatheticmoose — APPROVED** [review]
Approved with some minor nits. None are blockers but I do suggest changing the sidebar ordering. 

Generally, can we please take a close look at these docs as the user flow is far from optimal and you get bounced back and forth between sections and pages.

File: docusaurus/docs/publish-a-plugin/distribute-a-plugin.md (hunk 5 above)
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/distribute-a-plugin.md:11]: Good idea. Added to both places.

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md (hunk 10 above)
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/sign-a-plugin.md:16]: Fixed.

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md (hunk 11 above)
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/sign-a-plugin.md:26]: Fixed.

File: docusaurus/docs/publish-a-plugin/distribute-a-plugin.md (hunk 6 above)
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/distribute-a-plugin.md:19]: Captured in https://github.com/grafana/plugin-tools/issues/360

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md (hunk 8 above)
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:41]: Captured in https://github.com/grafana/plugin-tools/issues/360

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md (hunk 12 above)
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/sign-a-plugin.md:83]: Captured in https://github.com/grafana/plugin-tools/issues/360

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md (hunk 7 above)
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:38]: Deleted. Follow up captured in https://github.com/grafana/plugin-tools/issues/360.

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md (hunk 9 above)
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/sign-a-plugin.md:79]: Done.

**Pull request merged**